crates/sui-types/src/effects/object_change.rs

```diff
@@ -66,6 +67,34 @@ pub enum ObjectIn {
     Exist((VersionDigest, Owner)),
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum AccmulatorOperation {
+    /// Merge the value into the accumulator.
+    Merge,
+    /// Split the value from the accumulator.
+    Split,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum AccumulatorValue {
+    IntegerTuple(u128, u128),
+    // New accumulator types can be added here.
+    // For custom or complex types, we can use a `Vec<u8>` to store the raw value.
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct AccumulatorWriteV1 {
+    /// The recipient of the accumulator.
+    pub recipient: SuiAddress,
+    /// The type of the accumulator.
+    pub accumulator_type: TypeTag,
```

Comment:
Note: the ultimate destination of the event is a dynamic field. in #20649, i did the id computation in Move prior to emitting the event. In this system it seems that we have to compute the id outside of Move.

Either way can work, and it is definitely convenient to have the original recipient and type in the effects.

However, I have wondered whether we ever might want "namespaced" balances in the future, i.e. balances identified by `hash(address || type || user-supplied key)` - doing the key computation in Move gives us more flexibility to extend it in the future. (Of course we could also do it outside of move by introducing `AccumulatorWriteV2`)

Reply:
I probably should include the accumulator root object ID here too.